ComponentKit/TransactionalDataSources/Common/CKTransactionalComponentDataSourceListener.h

```diff
@@ -0,0 +1,30 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <Foundation/Foundation.h>
+
+@class CKTransactionalComponentDataSource;
+@class CKTransactionalComponentDataSourceAppliedChanges;
+@class CKTransactionalComponentDataSourceState;
+
+@protocol CKTransactionalComponentDataSourceListener
+
+/**
+ Announced on the main thread when the data source has just updated its state.
+ @param dataSource The sending data source; its state property now returns an updated object.
```

Comment:
You could vend the new state object, but it's redundant since you can fetch it from the data source.

We could //not// announce the datasource, but I'm not a fan of that because if some object is listening to multiple data sources, it can't distinguish the announcements it's receiving. We had this exact issue when @modocache tried to remove the "store" parameter from the Connection Store's announcements.

So in sum: this is the right thing because it matches Connection Store.